### PR TITLE
[AAP-8158] Add Credentials > Template List

### PR DESCRIPTION
**Pull request by daphnemaeve**
Jira ticket: https://issues.redhat.com/browse/AAP-8158

Adds Job Templates tab to Credentials page. Shows Job Templates using the selected Credentials.

Before:
Screenshot 2024-02-02 at 12 47 21 PM

After:
Screenshot 2024-02-02 at 12 46 59 PM

### CODE REVIEW

File: frontend/awx/access/credentials/CredentialPage/CredentialPage.tsx [hunk 1]
```diff
@@ -15,6 +15,7 @@ import { awxAPI } from '../../../common/api/awx-utils';
 import { Credential } from '../../../interfaces/Credential';
 import { AwxRoute } from '../../../main/AwxRoutes';
 import { useCredentialActions } from '../hooks/useCredentialActions';
+//import { CredentialJobTemplates } from './CredentialJobTemplates';
```
Comment (kialam) [inline frontend/awx/access/credentials/CredentialPage/CredentialPage.tsx:18]: Remove commented out line

File: frontend/awx/resources/templates/TemplatesList.tsx [hunk 2]
```diff
@@ -27,13 +27,18 @@ export function TemplatesList(props: {
...
+  credentialsId?: string | undefined;
 }) {
   const { t } = useTranslation();
   const pageNavigate = usePageNavigate();
   const getPageUrl = useGetPageUrl();
   const toolbarFilters = useTemplateFilters();
   const tableColumns = useTemplateColumns();
-  const getQueryParams = (projectId?: string | undefined, inventoryId?: string | undefined) => {
+  const getQueryParams = (
+    projectId?: string | undefined,
```
Comment (kialam) [inline frontend/awx/resources/templates/TemplatesList.tsx:38]: `undefined` type shouldn't be necessary if prop is optional.

**Review by kialam — APPROVED** [review]
A couple small things, otherwise this looks good. Pulled branch down and tested locally.

File: frontend/awx/resources/templates/TemplatesList.tsx (hunk 2 above)
Comment (daphnemaeve) [reply frontend/awx/resources/templates/TemplatesList.tsx]: Fixed! Also removed an unnecessary undefined above this in the TemplatesList props as well.

**Pull request merged**